packages/evm/contracts/Enclave.sol

```diff
@@ -232,7 +228,7 @@ contract Enclave is IEnclave, OwnableUpgradeable {
             InputDeadlinePassed(e3Id, e3.expiration)
         );
 
-        e3.inputValidator.enforce(msg.sender, data, Check.MAIN);
+        e3.inputValidator.validate(msg.sender, data);
```

Comment:
that's exactly my concern here - I think you should call `enforce()` and not `validate()` (which should not exist as second-way to enforcing the checks) here.